### PR TITLE
Runbook Command: lookup tenant by Id or name

### PR DESCRIPTION
**Pull request by paulegradie**
Previously we only allowed tenants to be discovered via ID. This change allows name or id to be used.

### CODE REVIEW

**Comment by paulegradie** [comment]
I applied rider's formatter to this file which change the formatting in a few places. Intros a bit of churn, I'll double check my formatting settings are correct in rider.

File: source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs [hunk 1]
```diff
@@ -116,7 +112,12 @@ public class RunRunbookCommand : ApiCommand, ISupportFormattedOutput
             options.Add<bool>("waitForRun", "[Optional] Whether to wait synchronously for deployment to finish.",
                 v => WaitForRun = true);
 
-            options.Add<bool>("progress", "[Optional] Show progress of the runbook run", v => { Progress = true; WaitForRun = true; NoRawLog = true; });
+            options.Add<bool>("progress", "[Optional] Show progress of the runbook run", v =>
```
Comment (andrewabest) [inline source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs:115]: For future reference, I think we should avoid too many formatting changes like this in PRs - it just muddies the waters when reviewing.

Comment (paulegradie) [reply source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs:115]: 👍 Agreed.

File: source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs [hunk 2]
```diff
@@ -197,11 +201,12 @@ protected override Task ValidateParameters()
...
 
             if ((RunAt ?? DateTimeOffset.Now) > NoRunAfter)
-                throw new CommandException("The Run will expire before it has a chance to execute.  Please select an expiry time that occurs after the deployment is scheduled to begin");
+                throw new CommandException(
+                    "The Run will expire before it has a chance to execute.  Please select an expiry time that occurs after the deployment is scheduled to begin");
 
             CheckForIntersection(IncludedMachineIds.ToList(), ExcludedMachineIds.ToList());
 
-            if (TenantIds.Contains("*") && (TenantIds.Count > 1 || TenantTagNames.Count > 0))
+            if (TenantNamesOrIds.Contains("*") && (TenantNamesOrIds.Count > 1 || TenantTagNames.Count > 0))
```
Comment (andrewabest) [inline source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs:209]: Just double checking - the server supports the wildcard alias already correct?

Comment (paulegradie) [reply source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs:209]: Last I checked (on friday). I'll check again right now.

Comment (paulegradie) [reply source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs:209]: Shizzy, thats, right -- this needs to be handled in the CLI now -- good catch good catch.

Comment (andrewabest) [reply source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs:209]: Well we need to pass through the wildcard if we get it, and not attempt to retrieve anything by name or id :)

Comment (paulegradie) [reply source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs:229]: 👍 Should be all set now. I added a method to apply this logic in the cli.

File: source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs [hunk 3]
```diff
@@ -179,6 +182,23 @@ public async Task Request()
             }
         }
 
+        private async Task<string[]> RetrieveTenants()
+        {
+
+            string[] tenants;
```
Comment (andrewabest) [inline source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs:188]:
```
return !TenantNamesOrIds.Contains("*")
    ? (await Repository.Tenants.FindByNamesOrIdsOrFail(TenantNamesOrIds).ConfigureAwait(false)).Select(x => x.Id).ToArray()
    : TenantNamesOrIds.ToArray()
```

File: source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs [hunk 4]
```diff
@@ -197,13 +217,14 @@ protected override Task ValidateParameters()
...
+                throw new CommandException(
+                    "The Run will expire before it has a chance to execute.  Please select an expiry time that occurs after the deployment is scheduled to begin");
 
             CheckForIntersection(IncludedMachineIds.ToList(), ExcludedMachineIds.ToList());
 
-            if (TenantIds.Contains("*") && (TenantIds.Count > 1 || TenantTagNames.Count > 0))
+            if (TenantNamesOrIds.Contains("*") && (TenantNamesOrIds.Count > 1 || TenantTagNames.Count > 0))
                 throw new CommandException(
-                    "When running on all tenants using the --tenantIds=* wildcard, no other tenant filters can be provided");
+                    "When running on all tenants using the --tenant=* wildcard, no other tenant filters can be provided");
```
Comment (andrewabest) [inline source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs:227]: Is there a test we need to update in line with the param name change, or has that already happened?

Comment (paulegradie) [reply source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs:227]: I think the param change was already done, but the exception message text hadn't been updated.

File: source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs (hunk 3 above)
Comment (paulegradie) [reply source/Octopus.Cli/Commands/Runbooks/RunRunbookCommand.cs:188]: 👍

**Pull request merged**